crates/loro-internal/src/encoding/encode_reordered.rs

```diff
@@ -2562,6 +2638,69 @@ mod arena {
         }
     }
 
+    #[derive(Clone)]
+    #[columnar(vec, ser, de, iterable)]
+    pub(super) struct PositionDelta<'a> {
+        // TODO: whether strategy
+        #[columnar(strategy = "Rle")]
+        common_prefix_length: usize,
+        #[columnar(borrow)]
+        rest: Cow<'a, [u8]>,
+    }
+
+    #[columnar(ser, de)]
+    pub(super) struct PositionArena<'a> {
+        #[columnar(class = "vec", iter = "PositionDelta<'a>")]
+        pub(super) positions: Vec<PositionDelta<'a>>,
+    }
+
+    impl<'a> PositionArena<'a> {
+        pub fn from_positions(positions: Vec<&'a [u8]>) -> Self {
```

Comment:
The encoded size can be much smaller if we sort the positions first.

We should use another kind of ValueRegister to handle that. i.e. It should preprocess all the posisition values and then sort the positions, and keep the mappings of `&[u8] => usize (position)`

Reply:
Could we reuse the `prop_that_used_for_sort` for sorting?  It is a enum instead of `i32`

Reply:
> It means collecting all position first, getting index later?

Yes

Reply:
done.

Reply:
What's the setup of the bench

Reply:
create 10000 nodes and randomly move them 100000 times